src/cim/components/ClusterDeployment/ClusterDeploymentWizard.tsx

```diff
@@ -33,18 +56,48 @@ const ClusterDeploymentWizard: React.FC<ClusterDeploymentWizardProps> = ({
 
     switch (stepId) {
       case 'hosts-selection':
-        return agentClusterInstall?.metadata?.name ? (
-          <ClusterDeploymentHostSelectionStep
-            clusterDeployment={clusterDeployment}
-            agentClusterInstall={agentClusterInstall}
-            onClose={onClose}
-            onSaveHostsSelection={onSaveHostsSelection}
-            agents={agents}
-            aiConfigMap={aiConfigMap}
-          />
-        ) : (
-          <LoadingState />
-        );
+        if (agentClusterInstall?.metadata?.name) {
+          if (isCIMFlow(clusterDeployment)) {
+            return (
+              <ClusterDeploymentHostSelectionStep
+                clusterDeployment={clusterDeployment}
+                agentClusterInstall={agentClusterInstall}
+                onClose={onClose}
+                onSaveHostsSelection={onSaveHostsSelection}
+                agents={agents}
+                aiConfigMap={aiConfigMap}
+              />
+            );
+          } else {
+            if (infraEnv) {
+              return (
+                <ClusterDeploymentHostsDiscoveryStep
+                  agentClusterInstall={agentClusterInstall}
+                  agents={aiFlowFilteredAgents}
```

Comment:
Do we need to filter agents by the label? Can't we just filter for agents which are bound to the cluster?

Reply:
It should be the same. But you are right, we can do the filtering one component level down.

Reply:
done